client/QuickStart/API_Util.cs

```diff
@@ -0,0 +1,410 @@
+        }
+
+        /// <summary>
+        /// Generates Kusto Connection String based on 'AppCertificate' Authentication Mode.
+        /// </summary>
+        /// <param name="clusterUrl">Url of cluster to connect to</param>
+        /// <param name="certificatePath">Given certificate path</param>
+        /// <param name="certificatePassword">Given certificate password</param>
+        /// <param name="applicationId">Given application id</param>
+        /// <param name="tenantId">Given tenant id</param>
+        /// <returns>AppCertificate Kusto Connection String</returns>
+        public static KustoConnectionStringBuilder CreateAppCertificateConnectionString(string clusterUrl, string certificatePath, string certificatePassword, string applicationId, string tenantId)
+        {
+            var missing = new[]
+            {
+                (name: nameof(certificatePath), value: certificatePath),
+                (name: nameof(certificatePassword), value: certificatePassword),
+                (name: nameof(applicationId), value: tenantId),
+            }.Where(item => string.IsNullOrWhiteSpace(item.value)).ToArray();
+            if (missing.Any())
+                ErrorHandler($"Missing the following required fields in configuration file in order to authenticate using a certificate: {string.Join(", ", missing.Select(item => item.name))}");
+
+            var appId = Environment.GetEnvironmentVariable("APP_ID");
+            var SubjectDistinguishedName = Environment.GetEnvironmentVariable("SUBJECT_DISTINGUISHED_NAME");
+            var IssuerDistinguishedName = Environment.GetEnvironmentVariable("ISSUER_DISTINGUISHED_NAME");
+            var privateKeyPemFilePath = Environment.GetEnvironmentVariable("PRIVATE_KEY_PEM_FILE_PATH");
+            var publicCertFilePath = Environment.GetEnvironmentVariable("PUBLIC_CERT_FILE_PATH");
+            string publicCertificate;
+            string pemCertificate;
+
+            try
+            {
+                pemCertificate = File.ReadAllText(privateKeyPemFilePath ?? throw new InvalidOperationException());
+            }
+            catch (InvalidOperationException e)
+            {
+                ErrorHandler($"Failed to load PEM file from {privateKeyPemFilePath}", e);
+            }
+
+            if (publicCertFilePath != null)
+            {
+                try
+                {
+                    publicCertificate = File.ReadAllText(publicCertFilePath);
+                }
+                catch (InvalidOperationException e)
+                {
+                    ErrorHandler($"Failed to load public certificate file from {publicCertFilePath}", e);
+                }
+                return new KustoConnectionStringBuilder(clusterUrl).WithAadApplicationSubjectAndIssuerAuthentication(appId, SubjectDistinguishedName, IssuerDistinguishedName, tenantId);
+            }
+
+
+            X509Certificate2 certificate = null;
```

Comment:
Remove - we don't want this here (right @yochai?)

Reply:
https://github.com/Azure/azure-kusto-samples-dotnet/pull/19#discussion_r884427197
converging with this discussion

Reply:
> Remove - we don't want this here (right @yochai?)

@AsafMah, you pinged the wrong user

Reply:
@yogilad